### PR TITLE
Details namespace fix

### PR DESCRIPTION
**Pull request by RickWinter**
Details namespace should be _details

### CODE REVIEW

File: docs/cpp/introduction.md [hunk 1]
```diff
@@ -175,10 +175,10 @@ The C++ SDK is designed for synchronous api calls.
 
 {% highlight cpp %}
 namespace Azure { namespace Group { namespace Service {
-namespace Details {
+namespace _details {
 // Part of the private API
 [[nodiscard]] int64_t ComputeHash(int32_t a, int32_t b) noexcept;
-} // namespace Details
+} // namespace _details
```
Comment (antkmsft) [inline docs/cpp/introduction.md:181]:
```suggestion
} // namespace _detail
```

File: docs/cpp/introduction.md [hunk 2]
```diff
@@ -194,7 +194,7 @@ namespace Details {
 
 {% include requirement/MUST id="cpp-design-naming-variables-constants" %} name namespace scope `const` or `constexpr` variables intended for user consumption with **PascalCase**.
 
-{% include requirement/MUST id="cpp-design-naming-variables-public-global" %} name namespace scope non-constant variables intended only for internal consumption with a `g_` prefix followed by **camelCase**. For example, `g_applicationContext`. Note that all such cases will be in a `Details` namespace or an unnamed namespace.
+{% include requirement/MUST id="cpp-design-naming-variables-public-global" %} name namespace scope non-constant variables intended only for internal consumption with a `g_` prefix followed by **camelCase**. For example, `g_applicationContext`. Note that all such cases will be in a `_details` namespace or an unnamed namespace.
```
Comment (antkmsft) [inline docs/cpp/introduction.md:197]:
```suggestion
{% include requirement/MUST id="cpp-design-naming-variables-public-global" %} name namespace scope non-constant variables intended only for internal consumption with a `g_` prefix followed by **camelCase**. For example, `g_applicationContext`. Note that all such cases will be in a `_detail` namespace or an unnamed namespace.
```

File: docs/cpp/introduction.md [hunk 3]
```diff
@@ -211,9 +211,9 @@ void Function(int parameterName) {
     int localName;
 }
 
-namespace Details {
+namespace _details {
```
Comment (antkmsft) [inline docs/cpp/introduction.md:214]:
```suggestion
namespace _detail {
```

File: docs/cpp/introduction.md [hunk 4]
```diff
@@ -211,9 +211,9 @@ void Function(int parameterName) {
...
     extern int g_internalUseGlobal;
-} // namespace Details
+} // namespace _details
```
Comment (antkmsft) [inline docs/cpp/introduction.md:216]:
```suggestion
} // namespace _detail
```

**Review by antkmsft — APPROVED** [review]
Please make a correction to use `_detail` instead of plural and then it looks good!

**Comment by RickWinter** [comment]
Its interesting to note that the cpp docs use "Details" in one place
https://github.com/MicrosoftDocs/cpp-docs/blob/main/docs/cpp/namespaces-cpp.md#nested-namespaces

**Comment by RickWinter** [comment]
Also to note the cpp reference spec does not prescribe a pattern from what I've seen.
https://en.cppreference.com/w/cpp/language/namespace

File: docs/cpp/introduction.md (hunk 1 above)
Comment (LarryOsterman) [reply docs/cpp/introduction.md:181]: What Anton said - the actual name is "_detail", similar to "_internal". We were wrong in the chat :(.

**Comment by LarryOsterman** [comment]
> Also to note the cpp reference spec does not prescribe a pattern from what I've seen. https://en.cppreference.com/w/cpp/language/namespace

100% agreed. The use of `Details`/`Internal`/`_internal`/`_detail` are local conventions to the C++ SDK.

Other local conventions: 
- types declared as `struct` should never have methods (@antkmsft enlightened me about that one yesterday - I disagree with this convention, but it does appear to be the case).
- Header file extensions are `.hpp` (rather than `.h`, `.hxx`, or `<no extension>`
- Implementation types are defined in headers named `Xxx_impl.hpp`
- Implementation types usually live in a directory named `private` under the implementation sources.
- Header files are located in a directory named something like:
  - `<root>/<service>/<package>/inc/azure/<package short name>/`
    - This causes the published header file for the package to be in `inc/azure/<package short name>/ ` on customer devices.

**Comment by antkmsft** [comment]
@LarryOsterman, there also seem to be an unwritten thing about the `_internal`/`_detail` namespaces - they (so far) can only be trailing namespaces, i.e. `Azure::Core::Http::_internal`, but not `Azure::Core::_internal::Http`.
There is also probably some de-facto pattern for where the `_internal`/`_detail` headers live, i.e. `azure/core/internal/xxx.hpp` etc.

@RickWinter, the conventions for `Details`/`_details`/`Detail`/`_detail` are per library conventions, there is no standardized naming. `detail` is somewhat common (boost, pytorch, opencv): https://www.bing.com/search?q=%22namespace+detail%22. Singular, but without underscore. We chose underscore so that it gets special treatment from IntelliSense, and does visually look like a "restricted area" - I think it is a good decision on our part to start with the underscore and use lowercase.

**Pull request merged**